aws-api/src/main/java/com/amplifyframework/api/aws/SingleItemResultOperation.java

```diff
@@ -132,33 +128,29 @@ public void onResponse(@NonNull Call call,
                 try {
                     jsonResponse = responseBody.string();
                 } catch (IOException exception) {
-                    responseListener.onError(new ApiException(
-                            "Could not retrieve the response body from the returned JSON",
-                            exception,
-                            AmplifyException.TODO_RECOVERY_SUGGESTION
+                    onFailure.accept(new ApiException(
+                        "Could not retrieve the response body from the returned JSON",
+                        exception, AmplifyException.TODO_RECOVERY_SUGGESTION
                     ));
+                    return;
```

Comment:
Nice catch!

Reply:
Nice catching on catching my catch! Haha, it was a big PR.